### PR TITLE
dev-nodejs/chalk and its dependencies

### PR DESCRIPTION
**Pull request by herbmillerjr**
What I read in the mailing list has basically said taking on packaging node modules is torture, but I'd still like to take it on. My endgame here is to eventually have all the nodejs modules I need to build Visual Studio Code OSS and bring it in.

The last commit, for chalk, adds the dev-nodejs category.

Not sure what to do about the x32 profiles repoman is complaining about:
  dependency.badindev           4
   dev-nodejs/color-name/color-name-1.1.3.ebuild: DEPEND: ~amd64(default/linux/amd64/17.0/x32)
['net-libs/nodejs']
   dev-nodejs/color-name/color-name-1.1.3.ebuild: RDEPEND: ~amd64(default/linux/amd64/17.0/x32)
['net-libs/nodejs']
   dev-nodejs/color-name/color-name-1.1.3.ebuild: DEPEND: ~amd64(hardened/linux/amd64/x32)
['net-libs/nodejs']
   dev-nodejs/color-name/color-name-1.1.3.ebuild: RDEPEND: ~amd64(hardened/linux/amd64/x32)
['net-libs/nodejs']

Related to bug: "V8 upstream said they won't support x32, bug #423815" But did that get resolved and nodejs never got unmasked for x32?

### CODE REVIEW

**Comment by mgorny** [comment]
Ahh, you're adding a new category, that's what confuses the hell out of assignment script. That said, I believe adding a new category requires a RFC to gentoo-dev@ first. Feel free to reference this PR as example of what would land in the new category.

**Comment by mgorny** [comment]
Oh, and new category must be the first commit, otherwise the added packages are invalid.

File: dev-nodejs/color-name/color-name-1.1.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+DEPEND="net-libs/nodejs"
+RDEPEND="${DEPEND}"
+
+src_install() {
+	insinto "/usr/$(get_libdir)/node_modules/${PN}"
+	doins -r *.js
```
Comment (mgorny) [inline dev-nodejs/color-name/color-name-1.1.3.ebuild]: Why `-r`? Are there `.js` directories in node?

File: dev-nodejs/color-name/color-name-1.1.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	doins package.json
+}
+
+src_test() {
+	/usr/bin/node test.js
```
Comment (mgorny) [inline dev-nodejs/color-name/color-name-1.1.3.ebuild]: You're missing error handling here (`|| die`).

File: dev-nodejs/color-name/metadata.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+		<name>Herb Miller Jr. (herbmillerjr)</name>
+	</maintainer>
+	<maintainer type="project">
+		<email>proxy-maint@gentoo.org</email>
+		<name>Proxy Maintainers</name>
+	</maintainer>
+	<upstream>
+		<remote-id type="github">colorjs/color-name</remote-id>
+	</upstream>
+</pkgmetadata>
```
Comment (mgorny) [inline dev-nodejs/color-name/metadata.xml]: Missing newline at the EOF.

**Comment by herbmillerjr** [comment]
I am having quite a bit of trouble getting subscribed to the gentoo-dev mailing list and posting. I'm not sure if new council decisions have anything to do with it:

https://bugs.gentoo.org/650964
https://bugs.gentoo.org/644070

I registered with the wrong e-mail address by accident. I can't seem to get registered with my main e-mail, and can't post from either. If I send you the message can you forward it to the list for me?

**Comment by mgorny** [comment]
We've gotten your mail just fine.

**Comment by mgorny** [comment]
I'm going to mark this red so that it isn't accidentally merged before people have a chance for feedback.

**Comment by flappyports** [comment]
No activity in 60+ days.

**Pull request closed without merging**